apps/studio/src/lib/db/clients/postgresql.ts

```diff
@@ -39,1347 +38,1419 @@ const pgErrors = {
-  const sql = `
-    SELECT datname
-    FROM pg_database
-    WHERE datistemplate = $1
-    ${databaseFilter ? `AND ${databaseFilter}` : ''}
-    ORDER BY datname
-  `;
+  if (orderBy && orderBy.length > 0) {
+    orderByString = "ORDER BY " + (orderBy.map((item) => {
+      if (_.isObject(item)) {
+        return `${wrapIdentifier(item.field)} ${item.dir.toUpperCase()}`
+      } else {
+        return wrapIdentifier(item)
+      }
+    })).join(",")
+  }
 
-  const params = [false];
+  if (_.isString(filters)) {
+    filterString = `WHERE ${filters}`
+  } else if (filters && filters.length > 0) {
+    let paramIdx = 1
+    const allFilters = filters.map((item) => {
+      if (item.type === 'in' && _.isArray(item.value)) {
+        const values = item.value.map((v, idx) => {
+          return options.inlineParams
+            ? knex.raw('?', [v]).toQuery()
+            : `$${paramIdx + idx}`
+        })
+        paramIdx += values.length
+        return `${wrapIdentifier(item.field)} ${item.type.toUpperCase()} (${values.join(',')})`
+      }
+      const value = options.inlineParams
+        ? knex.raw('?', [item.value]).toQuery()
+        : `$${paramIdx}`
+      paramIdx += 1
+      return `${wrapIdentifier(item.field)} ${item.type.toUpperCase()} ${value}`
+    })
+    filterString = "WHERE " + joinFilters(allFilters, filters)
 
-  const data = await driverExecuteSingle(conn, { query: sql, params });
+    params = filters.flatMap((item) => {
+      return _.isArray(item.value) ? item.value : [item.value]
+    })
+  }
 
-  return data.rows.map((row) => row.datname);
-}
+  const selectSQL = `SELECT ${selects.join(', ')}`
+  const baseSQL = `
+    FROM ${wrapIdentifier(options.schema)}.${wrapIdentifier(options.table)}
+    ${filterString}
+  `
+  // This comes from this PR, it provides approximate counts for PSQL
+  // https://github.com/beekeeper-studio/beekeeper-studio/issues/311#issuecomment-788325650
+  // however not using the complex query, just the simple one from the psql docs
+  // https://wiki.postgresql.org/wiki/Count_estimate
+  // however it doesn't work in redshift or cockroach.
+  const tuplesQuery = `
 
-export async function getInsertQuery(conn: HasPool, database: string, tableInsert: TableInsert): Promise<string> {
-  const columns = await listTableColumns(conn, database, tableInsert.table, tableInsert.schema)
+  SELECT
+    reltuples as total
+  FROM
+    pg_class
+  where
+      oid = '${wrapIdentifier(options.schema)}.${wrapIdentifier(options.table)}'::regclass
+  `
 
-  return buildInsertQuery(knex, tableInsert, columns, _.toString)
-}
+  // if we're not filtering data we want the optimized approximation of row count
+  // rather than a legit row count.
+  let countQuery = options.version.isPostgres && !filters && !options.forceSlow ? tuplesQuery : `SELECT count(*) as total ${baseSQL}`
+  if (options.version.isRedshift && !filters) {
+    countQuery = `SELECT COUNT(*) as total ${baseSQL}`
```

Comment:
Probably make a method for `countQuery` and then override `countQuery` in other postgres classes, or anything to avoid using `version.isPostgres`.